writeToStream.js

```diff
@@ -0,0 +1,579 @@
+
+'use strict';
+
+var protocol = require('./constants')
+  , empty = new Buffer(0)
+  , zeroBuf = new Buffer([0])
+  , numCache = require('./numbers')
+
+function generate(packet, stream) {
+
```

Comment:
@trevnorris thanks so much for providing context here! Would you say that using `nextTick()` is the best way for using `uncork()`? If so, I think we can just put this in and let the users of this module forget about it.

I would also protect it for backward compatibility:

``` js
    if (stream.cork) {
     stream.cork();
     process.nextTick(uncork, stream);
    }
    generate(packet, stream);
```